docs/nextConfigBaseline.js

```diff
@@ -0,0 +1,62 @@
+let DEPLOY_ENV = 'development';
+
+// Same as process.env.PULL_REQUEST_ID
+if (process.env.CONTEXT === 'deploy-preview') {
+  DEPLOY_ENV = 'pull-request';
+}
+
+if (process.env.CONTEXT === 'production' || process.env.CONTEXT === 'branch-deploy') {
+  DEPLOY_ENV = 'production';
+}
+
+if (
+  process.env.CONTEXT === 'branch-deploy' &&
+  (process.env.HEAD === 'master' || process.env.HEAD === 'next')
+) {
+  DEPLOY_ENV = 'staging';
+}
+
+process.env.DEPLOY_ENV = DEPLOY_ENV;
+
+function nextConfigBaseline(nextConfig) {
```

Comment:
I think there's a bit of an unwritten convention to start the name of those Next.js "plugins" `with`. e.g. `withNextConfigBaseline`, `withMuiBaseDocs`,...

Reply:
OK, I will use 

```suggestion
function withDocsInfra(nextConfig) {
```